### PR TITLE
Map Feature

### PR DESCRIPTION
**Pull request by chanmSC**
Created an admin Map page that can tag & preview locations on a map, to generate an array of pixel co-ordinates that can be copied into a public Map "puzzle" page (and users can access the updated map from the generated QR code).

Also created the public Map "puzzle" page to display the pixel co-ordinates entered in the answer field.

(Note:
I've added a new API endpoint for get_map_locations() to return the answer for the map puzzle to the public page on GET. I wanted to make sure we hardcoded an endpoint that only works for a puzzle named "map".

Feel free to restructure it properly by giving it its own API folder or db table, I didn't want to change the API code/admin/db code too much myself.)

Edit: Since I have access/permissions to this repo now, I've move the map branch from my forked repo to a normal branch in this repo!

### CODE REVIEW

**Comment by chanmSC** [comment]
Updated with the new function as a default game page (and not a puzzle page), and it loads in all the locations from puzzle.location to display to public map!

File: openday_scavenger/api/map/service.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+    Returns:
+        List of map location objects entered as the location of puzzles
+    """
+    # Get "map" puzzle answer which contains the pixel co-ordinates of puzzle locations
+    puzzles = get_all(db_session=db_session, only_active=True)
+
+    location_arrays = [p.location for p in puzzles if p.location]
+
+    locations: MapLocations = []
```
Comment (stmudie) [inline openday_scavenger/api/map/service.py]: I think this should be `locations: list[MapLocations] = []

Comment (stmudie) [reply openday_scavenger/api/map/service.py]: Or maybe not - still looking at this.

Comment (stmudie) [reply openday_scavenger/api/map/service.py]: No. Maybe it should be this:
```
puzzles = get_all(db_session=db_session, only_active=True)
    
location_arrays = [p.location for p in puzzles if p.location]

locations: MapLocations = MapLocations(locations=[])
for l_array in location_arrays:
    puzzle_locations = json.loads(l_array)
    for loc in puzzle_locations:
        locations.locations.append(loc)

return locations
```

Comment (stmudie) [reply openday_scavenger/api/map/service.py]: Except that breaks it. So... ```shrug```.

Comment (chanmSC) [reply openday_scavenger/api/map/service.py]: Updated, can you see if you still see the error?

**Review by stmudie — APPROVED** [review]
This is a fun feature!!!

**Pull request merged**